src/main/java/ch/sectioninformatique/auth/security/Role.java

```diff
@@ -89,16 +100,29 @@ public class Role {
     private Date updatedAt;
 
     /**
-     * Set of users who have this role.
+     * Set of users who have this mainRole.
+     * This field:
+     * - Implements a one-to-many relationship with User entity
+     * - Is mapped by the 'roles' field in the User class
+     * - Uses eager fetching to ensure roles are always available
+     * - Is ignored during JSON serialization to prevent infinite recursion
+     * - Is initialized as an empty HashSet to prevent null pointer exceptions
+     */
+    @OneToMany(mappedBy = "mainRole", fetch = FetchType.EAGER)
+    @JsonIgnore
+    private Set<User> usersMains = new HashSet<>();
+
+    /**
+     * Set of users who have this appspecificRole.
      * This field:
      * - Implements a many-to-many relationship with User entity
      * - Is mapped by the 'roles' field in the User class
      * - Uses eager fetching to ensure roles are always available
      * - Is ignored during JSON serialization to prevent infinite recursion
      * - Is initialized as an empty HashSet to prevent null pointer exceptions
      */
-    @ManyToMany(mappedBy = "roles", fetch = FetchType.EAGER)
+    @ManyToMany(mappedBy = "appSpecificRoles", fetch = FetchType.EAGER)
```

Comment:
App specific roles should only appear in the specific application and not in the spring-auth application (to be discussed).